charts/orchestrator/templates/_helpers.tpl

```diff
@@ -120,4 +120,25 @@
         {{- end -}}
         {{- .argoCDNamespace -}}
     {{- end -}}
+{{- end -}}
+
+{{- define "get-cluster-version" -}}
+  # Retrieve OCP version from `clusterversion` object
+  {{- $version :=(lookup "config.openshift.io/v1" "ClusterVersion" "" "version") }}
+  {{- range $version.status.history }}
+    {{- if eq .state "Completed" }}
+      {{- $v:=(semver .version) }}
+      {{- printf "OCP version is %d.%d" $v.Major $v.Minor }}
+    {{- end }}
+  {{- end }}
+
+  {{- $validMinors := list "13" "14" "15" "16" -}}
+  {{- if ne $v.Major "4" -}}
+      {{- fail (printf "Unsupported Major version: %s. Required: 4." $v.Major) -}}
+  {{- else if not (has $v.Minor $validMinors) -}}
```

Comment:
I think there's no need for `else`: if the previous conditional triggers the `fail` helm will exit execution.

Reply:
but if it won't, we should check also the minor version. e.g. version 4.10 will pass the first `if` but should fail for the `else` part.

Reply:
Agreed that it needs to check both. I'm just saying that if the first one triggers the `fail()` function, helm will exit execution.

Have you checked if [this would work,](https://helm.sh/docs/chart_template_guide/function_list/#semvercompare) assuming that you return the semver object from the call and not a map?

```
{{- if not (semverCompare ">=4.13 <=4.16" $v) -}}
  {{- fail (printf "Unsupported OCP version: %s.%s" $v.Major $v.Minor) -}}
{{- end -}}
```